src/phylum/ci/ci_base.py

```diff
@@ -454,18 +506,28 @@ def analyze(self) -> None:
         if self.all_deps:
             LOG.info("Considering all current dependencies ...")
             base_pkgs = []
-            total_packages = {pkg for depfile in self.depfiles for pkg in depfile.current_depfile_packages()}
+            total_packages = {pkg for depfile in self.depfiles for pkg in depfile.current_deps}
             LOG.debug("%s unique current dependencies", len(total_packages))
         else:
             LOG.info("Only considering newly added dependencies ...")
-            # When the `--force-analysis` flag is specified without the `--all-deps` flag, it is
-            # necessary to ensure the `is_depfile_changed` property is set for each dependency file.
-            # Simply referencing the `is_any_depfile_changed` property will ensure this happens.
-            if self.force_analysis and self.is_any_depfile_changed:
-                LOG.debug("Updated each dependency file's change status")
+            if self.force_analysis:
+                # When the `--force-analysis` flag is specified without the `--all-deps` flag, it is
+                # necessary to ensure the `is_depfile_changed` property is set for each dependency file.
+                # Simply referencing the `is_any_depfile_changed` property will ensure this happens.
+                LOG.debug("Updating each dependency file's change status ...")
+                if self.is_any_depfile_changed:
+                    LOG.debug("A provided/detected dependency file has changed")
+                else:
+                    LOG.debug("No provided/detected dependency file has changed")
             base_pkgs = sorted({pkg for depfile in self.depfiles for pkg in depfile.base_deps})
-            new_packages = sorted({pkg for depfile in self.depfiles for pkg in depfile.new_deps})
-            LOG.debug("%s unique newly added dependencies", len(new_packages))
+            new_pkgs = sorted({pkg for depfile in self.depfiles for pkg in depfile.new_deps})
+            LOG.debug("%s unique newly added dependencies", len(new_pkgs))
+            if not new_pkgs and any(isinstance(depfile, Manifest) for depfile in self.depfiles):
+                msg = """\
+                    No new dependencies were found. Is this a library?
+                      If so, consider specifying the `--all-deps` flag to
+                      ensure updated dependencies are taken into account."""
+                LOG.warning(textwrap.dedent(msg))
```

Comment:
It seems crazy to warn simply because nothing has changed... That will happen in any PR that doesn't change the dependencies...

I agree that the `--all-deps` flag is a good idea (especially when using manifest files). But that is true regardless of whether or not the current PR modifies the dependencies.

Reply:
It will be removed.